tests/camb/test_wrapper.py

```diff
@@ -151,3 +151,20 @@ def test_angular_diameter_distance(z, cosmo, compare):
         compare.angular_diameter_distance_z1z2(z1, z2).value,
         rtol=1e-3,
     )
+
+
+def test_transverse_comoving_distance(z, cosmo, compare):
+    np.testing.assert_allclose(
+        cosmo.transverse_comoving_distance(z),
+        compare.comoving_transverse_distance(z).value,
+        rtol=1e-4,
+    )
+
+    z1, z2 = z[:-1], z[1:]
+
+    np.testing.assert_allclose(
+        cosmo.transverse_comoving_distance(z1, z2),
+        compare.comoving_transverse_distance(z2).value
+        - compare.comoving_transverse_distance(z1).value,
+        rtol=0.3,
```

Comment:
Are we happy with this tolerance?

Reply:
No! 🤣

Reply:
Though presumably it's because the Zs are very close together? What's the atol?

Reply:
atol: 36.11869112
rtol: 0.20651554

Reply:
Maybe try spacing the zs further apart?

Reply:
Feels like we'll need to do a fair bit more fiddling with the `z` generation. I was hesitant to adapt the testing too much. Could you have a play to see what works well?

Reply:
Here's a fixture you can use:

```python
@pytest.fixture
def spaced_z(rng):
    r"""Return $z \in [0, 500]$ where $z_{i+1} - z_{i} >= 1$."""
    # Settings
    low, high = 0, 500
    min_spacing = 1
    n = 100

    # Generate random z values
    total_gap = (high - low) - min_spacing * (n - 1)
    assert total_gap > 0, "Total gap must be positive"
    # sample random "extra" gaps to add between each pair
    extra_gaps = rng.dirichlet(np.ones(n - 1)) * total_gap
    gaps = min_spacing + extra_gaps
    # cumulative sum to get positions
    return low + np.concatenate([[0], np.cumsum(gaps)])
```

Reply:
Gave it a go, sadly that was worse if anything (same order of magnitude)